main/lsp/requests/code_action.cc

```diff
@@ -25,30 +25,40 @@ LSPResult LSPLoop::handleTextDocumentCodeAction(unique_ptr<core::GlobalState> gs
     // Simply querying the file in question is insufficient since indexing errors would not be detected.
     auto run = tryFastPath(move(gs), files);
 
-    for (auto &e : run.errors) {
-        if (!e->isSilenced && e->loc.file() == file && !e->autocorrects.empty() &&
-            cmpRanges(*loc2Range(*run.gs, e->loc), *params.range) == 0) {
-            vector<unique_ptr<TextEdit>> edits;
-            edits.reserve(e->autocorrects.size());
-            for (auto &a : e->autocorrects) {
-                edits.emplace_back(make_unique<TextEdit>(loc2Range(*run.gs, a.loc), a.replacement));
+    for (auto &error : run.errors) {
+        if (!error->isSilenced && !error->autocorrects.empty()) {
+            // We return code actions corresponding to any error that encloses the request's range. Matching request
+            // ranges against error ranges exactly prevents VSCode's quick fix shortcut (Cmd+.) from matching any
+            // actions since it sends a 0 length range (i.e. the cursor). VSCode's request does include matching
+            // diagnostics in the request context that could be used instead but this simpler approach should suffice
+            // until proven otherwise.
+            if (!error->loc.contains(*range2Loc(*run.gs, *params.range.get(), file))) {
```

Comment:
There's only one `params.range`; can you run `range2Loc` outside of the loop and store it into a variable?